### PR TITLE
chore(sms): Remove sms code

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We have disabled sms on the frontend for a long time

## This pull request

- Removes our sms api, frontend pages and tests
- ~~Removes our signinCode (magic code) that is dependent on sms~~ I'll remove this in a follow up

## Issue that this pull request solves

Closes: https://github.com/mozilla/fxa/issues/11943

## Checklist

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/sms.js [hunk 1]
```diff
@@ -1,147 +0,0 @@
...
-  if (!config.sms.enabled) {
-    return [];
-  }
-
-  const REGIONS = new Set(config.sms.countryCodes);
-  const IS_STATUS_GEO_ENABLED = config.sms.isStatusGeoEnabled;
-
-  return [
-    {
-      method: 'POST',
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/routes/sms.js:26]: Just curious - when working on the Swagger API documentation, I noticed that there were a few routes that are noted as deprecated in the repo and current API doc (e.g., `/account/unlock/resend_code`). Would we want to note that the routes for sms and signinCodes are deprecated for the API docs too?

Comment (vbudhram) [reply packages/fxa-auth-server/lib/routes/sms.js:26]: @xlisachan Yea we can. I thought about returning an error allow those lines but figured it made more sense to rip it out instead lol.

File: packages/fxa-auth-server/lib/routes/signin-codes.js [hunk 2]
```diff
@@ -1,102 +0,0 @@
...
-        let base64 = request.payload.code.replace(/-/g, '+').replace(/_/g, '/');
-
-        const padCount = base64.length % 4;
-        for (let i = 0; i < padCount; ++i) {
-          base64 += '=';
-        }
-
-        const hex = Buffer.from(base64, 'base64').toString('hex');
-
-        const result = await db.consumeSigninCode(hex);
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/signin-codes.js:53]: It seems like outside of tests, this is the only place db.consumeSigninCode is used. Should the method be removed from the db module?

File: packages/fxa-auth-server/lib/routes/signin-codes.js [hunk 3]
```diff
@@ -1,102 +0,0 @@
...
-          },
-        },
-      },
-      handler: async function (request) {
-        log.begin('signinCodes', request);
-
-        const sessionToken = request.auth.credentials;
-
-        const metricsContext = await request.gatherMetricsContext({});
-        const signinCode = await db.createSigninCode(
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/signin-codes.js:84]: Same point as above. Is createSigninCode still needed in the db module?

File: packages/fxa-auth-server/lib/routes/sms.js [hunk 4]
```diff
@@ -1,147 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-'use strict';
-
-const error = require('../error');
-const isA = require('@hapi/joi');
-const PhoneNumberUtil = require('google-libphonenumber').PhoneNumberUtil;
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/sms.js:9]: Seems like this dependency is no longer needed. Remove from package.json?

File: packages/fxa-shared/test/metrics/amplitude.js [hunk 5]
```diff
@@ -318,13 +317,13 @@ describe('metrics/amplitude:', () => {
...
       before(() => {
-        result = transform({ type: 'sms.ios' }, {});
+        result = transform({ type: 'pair.ios' }, {});
       });
 
       it('returned the correct event data', () => {
         assert.equal(result.event_type, 'fxa_connect_device - cadEvent');
         assert.deepEqual(result.event_properties, {
-          connect_device_flow: 'sms',
+          connect_device_flow: 'pairing',
```
Comment (dschom) [inline packages/fxa-shared/test/metrics/amplitude.js:326]: Just out of curiousity, are these flow names/ids defined or documented somewhere?

**Review by dschom — COMMENTED** [review]
As I was checking references in the code, I found a couple more places where maybe there could be some more deletions. Here’s a list with permalinks:

template.includes('sms’), I don’t think any of them do anymore
https://github.com/mozilla/fxa/blob/858e1dfd5e80ebf175087528953d6bdfaf7a994e/packages/fxa-auth-server/test/scripts/write-emails-to-disk.js#L39
  error.viewName:
https://github.com/mozilla/fxa/blob/0a198a76a4fd6f776407e5d097ba0aa3b7f16af7/packages/fxa-content-server/app/tests/spec/lib/metrics.js#L643
https://github.com/mozilla/fxa/blob/0a198a76a4fd6f776407e5d097ba0aa3b7f16af7/packages/fxa-content-server/app/tests/spec/lib/metrics.js#L691


ClientApi.prototype.*SigninCode, did these endpoints get removed?
https://github.com/mozilla/fxa/blob/fd3f7209ae0bd1985d47cfcc254159953f5e695c/packages/fxa-auth-server/test/client/api.js#L943
https://github.com/mozilla/fxa/blob/fd3f7209ae0bd1985d47cfcc254159953f5e695c/packages/fxa-auth-server/test/client/api.js#L950
https://github.com/mozilla/fxa/blob/fd3f7209ae0bd1985d47cfcc254159953f5e695c/packages/fxa-auth-server/test/client/index.js#L793
https://github.com/mozilla/fxa/blob/fd3f7209ae0bd1985d47cfcc254159953f5e695c/packages/fxa-auth-server/test/client/index.js#L797

Outdated doc: https://github.com/mozilla/fxa/blob/c14d4aac4e083f21679931b3f65ff7f3f74ad1f1/packages/fxa-content-server/app/scripts/lib/metrics.js#L666
 Reference to sms end point in test name… minor nitpic: https://github.com/mozilla/fxa/blob/fd3f7209ae0bd1985d47cfcc254159953f5e695c/packages/fxa-content-server/app/tests/spec/models/reliers/browser.js#L152-L153

Adde note about deprecated error code? Seems like this file and auth-errors.ts in fxa-settings are aligned. https://github.com/mozilla/fxa/blob/47074a15060224b497725d9c5f9c93e5913378f0/packages/fxa-content-server/app/scripts/lib/auth-errors.js#L127
 Is signInCodes still a valid feature? https://github.com/mozilla/fxa/blob/d584a10bcad53eb0c6fd5b5eb618161a472e0e6c/packages/fxa-auth-server/test/local/server.js#L171
https://github.com/mozilla/fxa/blob/d584a10bcad53eb0c6fd5b5eb618161a472e0e6c/packages/fxa-auth-server/test/local/server.js#L295
https://github.com/mozilla/fxa/blob/d584a10bcad53eb0c6fd5b5eb618161a472e0e6c/packages/fxa-auth-server/test/local/server.js#L207
https://github.com/mozilla/fxa/blob/d584a10bcad53eb0c6fd5b5eb618161a472e0e6c/packages/fxa-auth-server/test/local/server.js#L425

There are still signingCodes in the db module. Here’s some test for them. Are these needed? https://github.com/mozilla/fxa/blob/fa817cb37809c622a9f313356b9dc17bdab265f8/packages/fxa-auth-server/test/remote/db_tests.js#L1373

File: packages/fxa-shared/test/metrics/amplitude.js (hunk 5 above)
Comment (clouserw) [reply packages/fxa-shared/test/metrics/amplitude.js:326]: not all of them. I'd love to get some docs sometime!

**Review by dschom — APPROVED** [review]
LGTM! Feedback concerning further deletions can be addressed in a follow up PR. I also did some exploratory testing that showed no cause for concern.

**Pull request merged**